src/include/duckdb/storage/block_manager.hpp

```diff
@@ -75,7 +76,19 @@ class BlockManager {
 
 	void UnregisterBlock(block_id_t block_id, bool can_destroy);
 
+	//! Returns a reference to the metadata manager of this block manager.
 	MetadataManager &GetMetadataManager();
+	//! Returns the block allocation size of this block manager.
+	//! Not to be confused with the block size.
+	inline idx_t GetBlockAllocSize() const {
+		return block_alloc_size;
+	}
+	//! Sets the block allocation size. This should only happen when initializing an existing database.
+	//! When initializing an existing database, we construct the block manager before reading the file header,
+	//! which contains the file's actual block allocation size.
+	void SetBlockAllocSize(const idx_t block_alloc_size_p) {
+		block_alloc_size = block_alloc_size_p;
```

Comment:
Maybe here we can check it is only set once, e.g. `if (block_alloc_size != 0) throw InternalException(...)`